### PR TITLE
Fix Leaflet panInside which supports PanOptions and ZoomOptions

### PR DESCRIPTION
**Pull request by bb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Leaflet/Leaflet/blob/master/src/map/Map.js#L536 panInside calls pan and forwards the options
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## leaflet ([<kbd>unpkg</kbd>](https://unpkg.com/browse/leaflet@latest/))
was missing the following properties:
1. Mixin
1. PosAnimation
1. geoJson
1. noConflict
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cf914c52d1413e935bfb144326e7eddce0fe6d9f
</p>

File: types/leaflet/index.d.ts [hunk 1]
```diff
@@ -1615,7 +1615,7 @@ export interface FitBoundsOptions extends ZoomOptions, PanOptions {
     maxZoom?: number | undefined;
 }
 
-export interface PanInsideOptions {
+export interface PanInsideOptions extends ZoomOptions, PanOptions {
```
Comment (domdomegg) [inline types/leaflet/index.d.ts:1618]:
```suggestion
export interface PanInsideOptions extends PanOptions {
```

panInside calls panTo which calls setView which takes pan and zoom options.

However, panTo does not pass through the zoom options: https://github.com/Leaflet/Leaflet/blob/d52d2fa341af4c120262b24a12592cfc3f66020e/src/map/Map.js#L308

**Review by domdomegg — CHANGES_REQUESTED** [review]
As per code comment

**Comment by bb** [comment]
Thanks for the correction, @domdomegg! I committed your suggestion.

**Review by domdomegg — APPROVED** [review]

**Review by henrythasler — APPROVED** [review]

**Comment by bb** [comment]
Ready to merge

**Pull request merged**